### ISSUE TITLE
Optional not supported by Proto3FileBuilder

### ISSUE THREAD

**Issue by Anmol92verma** [TimOrtel/GRPC-Kotlin-Multiplatform#5]
When used optional it throws RuntimeException "No message or enum $typeText found."

**Comment by TimOrtel** [comment]
Yes, there is no support for optional fields. What is your usecase for optional?

**Comment by TimOrtel** [comment]
Can you please give an example proto file and the explanation on where and why you use the optional keyword.

**Comment by TimOrtel** [comment]
The link you sent me is for proto2. However, this project targets proto3.

**Comment by Anmol92verma** [comment]
Proto 3 also supports optional right ?

**Comment by TimOrtel** [comment]
All fields are by default optional in Proto3.

**Comment by Anmol92verma** [comment]
Problem is when i use the km* dsl from the library they are marked as NonNull i guess.

**Comment by TimOrtel** [comment]
If you do not want it to be set, do not set it in the builder. If you want to know if it is set, use the has*() function.
